spec/controllers/overrides/passwords_controller_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+describe Overrides::PasswordsController do
+  before(:each) do
+    @request.env['devise.mapping'] = Devise.mappings[:user]
+  end
+
+  let!(:user) do
+    @user ||= User.create(
+      email: 'jane@doe.com',
+      password: 'somepassword',
+      password_confirmation: 'somepassword',
+    )
+  end
+
+  describe '#create' do
+    it 'should send an email' do
+      expect_any_instance_of(User).to receive(:send_devise_notification).and_return(true)
+      post :create, email: user.email, redirect_url: 'http://localhost:3000/forgot-password'
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe '#edit' do
+    let(:token) do
+      allow_any_instance_of(User).to receive(:send_devise_notification).and_return(true)
+      @token ||= user.send_reset_password_instructions
+    end
+
+    it 'change allow_password_change to true' do
```

Comment:
i'm a little confused about the logic around allow_password_change, why do we set it to true on edit?

Reply:
This is devise behavior. We are expecting it to be set to true. I presume it has something to do with safety. This functionality is available for unsigned users, so without token unuathorized user is not able to change password for some other account.